pkg/asset/installconfig/azure/validation_test.go

```diff
@@ -127,6 +128,59 @@ var (
 			},
 		},
 	}
+
+	diskEncryptionSetID          = "test-encryption-set-id"
+	diskEncryptionSetName        = "test-encryption-set-name"
+	diskEncryptionSetType        = "test-encryption-set-type"
+	diskEncryptionSetLocation    = "disk-encryption-set-location"
+	validDiskEncryptionSetResult = &azenc.DiskEncryptionSet{
+		ID:       to.StringPtr(diskEncryptionSetID),
+		Name:     to.StringPtr(diskEncryptionSetName),
+		Type:     to.StringPtr(diskEncryptionSetType),
+		Location: to.StringPtr(diskEncryptionSetLocation),
+	}
+
+	validDiskEncryptionSetSubscriptionID = "test-encryption-set-subscription-id"
+	validDiskEncryptionSetResourceGroup  = "test-encryption-set-resource-group"
+	validDiskEncryptionSetName           = "test-encryption-set-name"
+	validDiskEncryptionSetConfig         = func() *azure.DiskEncryptionSet {
+		return &azure.DiskEncryptionSet{
+			SubscriptionID: validDiskEncryptionSetSubscriptionID,
+			ResourceGroup:  validDiskEncryptionSetResourceGroup,
+			Name:           validDiskEncryptionSetName,
+		}
+	}
+
+	validDiskEncryptionSetControlPlane = func(ic *types.InstallConfig) {
+		ic.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet = validDiskEncryptionSetConfig()
+	}
+	validDiskEncryptionSetCompute = func(ic *types.InstallConfig) {
+		ic.Compute[0].Platform.Azure.OSDisk.DiskEncryptionSet = validDiskEncryptionSetConfig()
+	}
+	invalidateDiskEncryptionSetControlPlaneSubscriptionID = func(ic *types.InstallConfig) {
+		ic.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet = validDiskEncryptionSetConfig()
+		ic.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet.SubscriptionID = ""
+	}
+	invalidateDiskEncryptionSetControlPlaneResourceGroup = func(ic *types.InstallConfig) {
+		ic.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet = validDiskEncryptionSetConfig()
+		ic.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet.ResourceGroup = ""
+	}
+	invalidateDiskEncryptionSetControlPlaneName = func(ic *types.InstallConfig) {
+		ic.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet = validDiskEncryptionSetConfig()
+		ic.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet.Name = ""
+	}
+	invalidateDiskEncryptionSetComputeSubscriptionID = func(ic *types.InstallConfig) {
+		ic.Compute[0].Platform.Azure.OSDisk.DiskEncryptionSet = validDiskEncryptionSetConfig()
+		ic.Compute[0].Platform.Azure.OSDisk.DiskEncryptionSet.SubscriptionID = ""
+	}
+	invalidateDiskEncryptionSetComputeResourceGroup = func(ic *types.InstallConfig) {
+		ic.Compute[0].Platform.Azure.OSDisk.DiskEncryptionSet = validDiskEncryptionSetConfig()
+		ic.Compute[0].Platform.Azure.OSDisk.DiskEncryptionSet.ResourceGroup = ""
+	}
+	invalidateDiskEncryptionSetComputeName = func(ic *types.InstallConfig) {
+		ic.Compute[0].Platform.Azure.OSDisk.DiskEncryptionSet = validDiskEncryptionSetConfig()
+		ic.Compute[0].Platform.Azure.OSDisk.DiskEncryptionSet.Name = ""
+	}
```

Comment:
Having these test variations on the different fields is confusing because it does not test code we wrote: these field variations are different ways for the API to fail in the same way. 

From what I see the test state matrix is 6 different states: 3 machine pools (master, worker, default) x 2 DiskEncryptionSetStates (found, not found). I could be convinced otherwise. For example, there might be a way to factor out the machine pools, but it is not obvious to me.

Reply:
Fixed.